tools/clang/include/clang/SPIRV/EmitVisitor.h

```diff
@@ -81,12 +82,10 @@ class EmitTypeHandler {
 /// representation.
 class EmitVisitor : public Visitor {
 public:
-  EmitVisitor(const SpirvCodeGenOptions &opts, SpirvContext &ctx,
-              SpirvModule &mod)
-      : Visitor(opts, ctx, mod) {
-    typeHandler = new (ctx)
-        EmitTypeHandler(ctx, mod, &annotationsBinary, &typeConstantBinary);
-  }
+  EmitVisitor(const SpirvCodeGenOptions &opts, SpirvContext &ctx)
+      : Visitor(opts, ctx), id(0),
+        typeHandler(ctx, &annotationsBinary, &typeConstantBinary,
+                    [this]() -> uint32_t { return takeNextId(); }) {}
```

Comment:
This is a little bit tricky here. `this` object is not fully initialized yet at the time of initializing `typeHandler`. Normally we should not do this, but since it's just capturing the pointer, maybe it's fine.

Reply:
Hmmm... We can switch to using a static method and not pass in any lambdas. Will consider for the future.